### DISCUSSION TITLE
In the official Azure Developer CLI templates, why are some outputs in UPPER_CASE and some in camelCase?

### DISCUSSION THREAD

**Discussion by kurt-mueller-osumc** [Azure/azure-dev#1774] [Q&A] [answered]
In the official `azd` templates, some outputs are in UPPER_CASE while others are in camelCase. Is there are a reason both are being used?

For example:
1. the outputs in https://github.com/Azure-Samples/todo-csharp-cosmos-sql/blob/main/infra/core/host/appservice.bicep are in camelCase
2. the outputs in https://github.com/Azure-Samples/todo-csharp-cosmos-sql/blob/main/infra/core/database/postgresql/flexibleserver.bicep are UPPER_CASE

Why use one over the other and in what situations?

**Comment by ellismg** [comment, marked as answer]
>  Is there are a reason both are being used?

When you run `azd provision`, `azd` will take all the outputs from your `main.bicep` file and write them into the environment file so they are available as environment variables in later operations (for example, when doing replacements in `main.parameters.json` or when running commands via our hooks system).

Because of that, we used the convention of `UPPER_CASE` for outputs in main.bicep because that matches the general convention folks use for environment variables.

The modules we have here actually evolved out of earlier iterations of our templates, where everything was done in `main.bicep` and as part of that evolution, we ended up just reusing this casing strategy.

This goes against the general grain of idiomatic bicep templates which typically use camelCase, given their heritage coming from ARM Templates, which are written in JSON and so camelCase feels idiomatic there.

Our long term plan is to take these modules and move them into the bicep registry.  I suspect as part of doing that work we'll end up having the outputs all move to camelCase (mirroring what we did with the parameters), to match the design language of other bicep modules.

So it's probably best to continue to use UPPER_CASE for outputs in your main.bicep file, since they will be available as environment variables in some cases and some platforms allow case sensitivity for environment variable names.  For everything else, it doesn't matter to `azd`, and using camelCase probably makes sense because it will just feel more like most of the other bicep code out there in the world.
